File: code/LeetCode/2320[DP].py
class Solution:
    def countHousePlacements(self, n: int) -> int:
        # f[i]=f[i-1]+1+f[i-2]
        MOD=10**9+7
        f=[0]*(n+2)
        for i in range(n):
            f[i+2]=(f[i+1]%MOD+1+f[i]%MOD)%MOD
        return ((f[-1]+1)**2)%MOD

File: CompetitiveProgramming/Old/LeetCode/2320[dp].cpp
class Solution {
public:
    int countHousePlacements(int n) {
        int dp[n+5];
        dp[0]=1;
        dp[1]=2;
        for(int i=2;i<n+2;i++){
            dp[i]=((long long)dp[i-2]+dp[i-1])%((int)1e9+7);
        }
        return ((long long)dp[n]*dp[n])%((int)1e9+7);
    }
};